component/http/middleware.go

```diff
@@ -158,21 +158,12 @@ func logRequestResponse(corID string, w *responseWriter, r *http.Request) {
 
 func getOrSetCorrelationID(h http.Header) string {
 	cor, ok := h[correlation.HeaderID]
-	if !ok {
-		corID := uuid.New().String()
-		h.Set(correlation.HeaderID, corID)
-		return corID
-	}
-	if len(cor) == 0 {
-		corID := uuid.New().String()
-		h.Set(correlation.HeaderID, corID)
-		return corID
-	}
-	if cor[0] == "" {
+	if !ok || len(cor) == 0 || cor[0] == "" {
```

Comment:
I don't mind leaving it as it was, tbh.
We can also decrease the number of if's by skipping `!ok` because in this particular case `!ok` and `len(cor) == 0` are equivalent. 
https://play.golang.org/p/JdbMC4h3Tzl